pkg/virtctl/create/instancetype/instancetype.go

```diff
@@ -0,0 +1,263 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package instancetype
+
+import (
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	v1 "kubevirt.io/api/core/v1"
+	instancetypev1alpha2 "kubevirt.io/api/instancetype/v1alpha2"
+	"sigs.k8s.io/yaml"
+
+	"kubevirt.io/kubevirt/pkg/virtctl/create/params"
+)
+
+const (
+	Instancetype = "instancetype"
+
+	CPUFlag             = "cpu"
+	MemoryFlag          = "memory"
+	GPUFlag             = "gpu"
+	HostDeviceFlag      = "hostdevice"
+	IOThreadsPolicyFlag = "iothreadspolicy"
+	NameFlag            = "name"
+	ClusterFlag         = "cluster"
+)
+
+type instancetype struct {
+	name   string
+	cpu    int
+	memory string
+
+	gpus            []string
+	hostDevices     []string
+	ioThreadsPolicy string
+	cluster         bool
+}
+
+type GPU struct {
+	Name       string `param:"name"`
+	DeviceName string `param:"devicename"`
+}
+
+type HostDevice struct {
+	Name       string `param:"name"`
+	DeviceName string `param:"devicename"`
+}
+
+var flags = []string{
```

Comment:
If the order in which the flags are processed is not relevant you can drop this additional slice and just iterate the `optFns` map.